orchestrator/models.py

```diff
@@ -1,28 +1,60 @@
 import uuid
 from enum import IntEnum
 from django.db import models
+from django.utils import timezone
 from django.contrib.postgres.fields import JSONField
 
 
 def message_default():
     message_default_dict = {
         'log': '',
         'failed_jobs': {},
-        'unknown_jobs':{},
+        'unknown_jobs': {},
         'info': ''
     }
     return message_default_dict
 
 
 class Status(IntEnum):
     CREATED = 0
-    PENDING = 1
-    RUNNING = 2
-    COMPLETED = 3
-    FAILED = 4
-    ABORTED = 5
-    UNKNOWN = 6
-    SUSPENDED = 7
+    SUBMITTING = 1
+    SUBMITTED = 2
+    PENDING = 3
+    RUNNING = 4
+    COMPLETED = 5
+    FAILED = 6
+    ABORTED = 7
+    UNKNOWN = 8
+    SUSPENDED = 9
+
+    def transition(self, transition_to):
+        if self == self.CREATED:
+            if transition_to in (self.SUBMITTING, self.ABORTED,):
+                return True
+        elif self == self.SUBMITTING:
+            if transition_to in (self.SUBMITTED, self.ABORTED):
+                return True
+        elif self == self.SUBMITTED:
+            if transition_to in (
+            self.PENDING, self.RUNNING, self.COMPLETED, self.FAILED, self.ABORTED, self.SUSPENDED, self.UNKNOWN):
```

Comment:
I think we should document a few of the statuses. I'm not entirely sure the difference between SUBMITTED/PENDING. And also explain why it's possible for SUBMITTED to skip PENDING/RUNNING -- which I assume has to do with the gap in LSF checks

Reply:
I'll add comments, and logging.